spec/models/grda_warehouse/tasks/system_maintenance_task_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe GrdaWarehouse::Tasks::SystemMaintenanceTask, type: :model do
+  let(:task) { create(:system_maintenance_task) }
+
+  describe 'associations' do
+    it { is_expected.to have_many(:system_maintenance_task_runs) }
+  end
+
+  describe '#threshold_exceeded?' do
+    let(:task) { create(:system_maintenance_task, alert_threshold_minutes: 60) }
+
+    context 'when there are no completed runs' do
+      it 'returns true' do
+        expect(task.threshold_exceeded?).to be true
+      end
+    end
+
+    context 'when there are recent completed runs within threshold' do
+      before do
+        create(:system_maintenance_task_run,
+               system_maintenance_task: task,
+               started_at: 30.minutes.ago,
+               completed_at: 25.minutes.ago)
+      end
+
+      it 'returns false' do
+        expect(task.threshold_exceeded?).to be false
+      end
+    end
+
+    context 'when the most recent completed run is outside threshold' do
+      before do
+        create(:system_maintenance_task_run,
+               system_maintenance_task: task,
+               started_at: 90.minutes.ago,
+               completed_at: 85.minutes.ago)
+      end
+
+      it 'returns true' do
+        expect(task.threshold_exceeded?).to be true
+      end
+    end
+
+    context 'when there are incomplete runs within threshold' do
+      before do
+        create(:system_maintenance_task_run,
+               system_maintenance_task: task,
+               started_at: 30.minutes.ago,
+               completed_at: nil)
+      end
+
+      it 'returns true' do
+        expect(task.threshold_exceeded?).to be true
+      end
```

Comment:
If this is only 30 minutes old, shouldn't it not trigger?  (Threshold is 60 right?)